util/chunk/iterator.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunk
+
+// Iterator is used to iterate a number of rows.
+//
+// for row := it.Begin(); row != it.End(); row = it.Next() {
+//     ...
+// }
+type Iterator interface {
+	// Begin resets the cursor of the iterator and returns the first Row.
+	Begin() Row
+
+	// Next returns the next Row.
+	Next() Row
+
+	// End returns the invalid end Row.
+	End() Row
+}
+
+// NewSliceIterator returns a Iterator for Row slice.
+func NewSliceIterator(rows []Row) Iterator {
+	return &sliceIterator{rows: rows}
+}
+
+type sliceIterator struct {
+	rows   []Row
+	cursor int
+}
+
+func (it *sliceIterator) Begin() Row {
+	if len(it.rows) == 0 {
+		return Row{}
```

Comment:
how about return `it.End()` once we reach the end of iterator during the execution of `Begin()` and `Next()`